web/modules/custom/signin_popup/src/Plugin/Block/JoinBlock.php

```diff
@@ -12,15 +16,64 @@
  *   admin_label = @Translation("Join now"),
  * )
  */
-class JoinBlock extends BlockBase {
+class JoinBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * @var AccountInterface $account
+   */
+  protected $account;
+
+  /**
+   * The language manager.
+   *
+   * @var \Drupal\Core\Language\LanguageManager
+   */
+  protected $languageManager;
+
+  /**
+   * @param array $configuration
+   * @param string $plugin_id
+   * @param mixed $plugin_definition
+   * @param \Drupal\Core\Session\AccountInterface $account
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, AccountInterface $account, LanguageManager $language_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->account = $account;
+    $this->languageManager   = $language_manager;
+  }
+
+  /**
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   * @param array $configuration
+   * @param string $plugin_id
+   * @param mixed $plugin_definition
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('current_user'),
+      $container->get('language_manager')
+    );
+  }
 
   /**
    * {@inheritdoc}
    */
   public function build() {
+    $roles = \Drupal::currentUser()->getRoles();
+    $is_editor = FALSE;
+    if (in_array('editor', $roles) || in_array('administrator', $roles)) {
```

Comment:
Instead of getting roles and compare with desired ones, it's better to remove [line 67](https://github.com/drupalAM/drupal-am/pull/123/files#diff-25506a7d6e3106893a7cf2622e797335bf6e44e8fa6441c500ed838179937f2bR67) and here just write:
`if ($this->account->hasRole('editor') || $this->account->hasRole('administrator')) {`